### ISSUE TITLE
Add support for repl

### ISSUE THREAD

**Issue by eatonphil** [multiprocessio/dsq#53]
Grab https://github.com/chzyer/readline and wrap main.go's query parts (likely line 469-503) in a loop. Allow the loop to keep going if the user passes an -i flag or --interactive flag.

When this flag is in, caching must also be turned on so that the sqlite database is not ephemeral per run (I think).

**Comment by Ghibranalj** [comment]
How do you like the behaviour to be?. like how to exit and stuff.

**Comment by eatonphil** [comment]
Control-D to exit for now.

And have `dsq >` be the prompt

**Comment by Ghibranalj** [comment]
How should the test be written?

**Comment by eatonphil** [comment]
Oh yeah I realize now you can't just wrap lines 469-503 in a loop. But you can pull them out into a runQuery helper and then call that helper within a repl loop if interactive mode is on

**Comment by Ghibranalj** [comment]
Okay on it

**Comment by Ghibranalj** [comment]
Are you comfortable with adding a dependency for reading the input.

I was able to implement basic REPL by using the built in bufio package but this doesnt support moving the cursor around with the arrow keys.

**Comment by eatonphil** [comment]
Yup, I mentioned a suggested dependency in the original message for this issue
